### PR TITLE
[Filter/NNFW] change nnfw api

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. Change nnfw api to set input info.
The function apply_tensorinfo will be deprecated soon.
2. Remove unnecessary definition about max rank of nnfw.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
NNFW 1.6.0 includes nnfw_set_input_tensorinfo(). need to merge this change later.

**Comment by myungjoo** [comment]
> NNFW 1.6.0 includes nnfw_set_input_tensorinfo(). need to merge this change later.

Then, can you add a static-version-checker? (does NNFW have it?)

**Comment by jaeyun-jung** [comment]
> > NNFW 1.6.0 includes nnfw_set_input_tensorinfo(). need to merge this change later.
> 
> Then, can you add a static-version-checker? (does NNFW have it?)

add todo and definition to build nnfw in tizen.

File: ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c [hunk 1]
```diff
@@ -396,10 +394,21 @@ nnfw_tensor_info_set (const nnfw_pdata * pdata,
   for (idx = nnfw_info.rank - 1; idx >= 0; idx--)
     nnfw_info.dims[nnfw_info.rank - idx - 1] = info->dimension[idx];
 
-  for (idx = NNFW_TENSOR_RANK_LIMIT - 1; idx >= nnfw_info.rank; idx--)
+  /** @note Maximum rank expressible with nnfw is 6 (NNFW_MAX_RANK) */
+  for (idx = NNFW_MAX_RANK - 1; idx >= nnfw_info.rank; idx--)
     nnfw_info.dims[idx] = 0;
 
-  nnfw_apply_tensorinfo (pdata->session, tensor_idx, nnfw_info);
+#if defined (__TIZEN__)
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c:401]: What about https://mesonbuild.com/Compiler-properties.html#does-a-function-exist ?

Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: add todo and new definition (NNFW_USE_OLD_API) for this case

**Pull request merged**